src/verifai/samplers/domain_sampler.py

```diff
@@ -25,6 +25,14 @@ class DomainSampler:
     def __init__(self, domain):
         self.domain = domain
 
+    def getSample(self):
+        """Generate the next sample, given the current distribution."""
+        return self.nextSample(feedback=None)
```

Comment:
Aren't `getSample` and `update` now the primary APIs, with `nextSample` only being used for backwards compatibility for sequential samplers? If so, I'd have `getSample` raise `NotImplementedError` and have `nextSample` be like this:
```
if self.last_sample is not None:
    assert feedback is not None
    self.update(self.last_sample, self.last_info, feedback)
self.last_sample, self.last_info = self.getSample()
return self.last_sample
```
This way, new samplers only need to implement `getSample` and `update`, but `nextSample` will still work if someone wants to use the old sequential API. Make sense?